src/runner.ts

```diff
@@ -116,17 +115,14 @@ export class Runner {
   private async _runTests(suite: RunnerSuite): Promise<RunResult> {
     // Trial run does not need many workers, use one.
     const runner = new Dispatcher(suite, { ...config, jobs: config.jobs }, this._reporter);
-    try {
-      for (const f of this._beforeFunctions)
-        await f();
-      this._reporter.onBegin(config, suite);
-      await runner.run();
-      await runner.stop();
-      this._reporter.onEnd();
-    } finally {
-      for (const f of this._afterFunctions)
-        await f();
-    }
+    let sigint = false;
+    const sigIntPromise = new Promise(f => process.on('SIGINT', () => { sigint = true; f(); }));
```

Comment:
nit: might as well remove the sigint handler here, if we aim at reusable runner class.